dali/kernels/test/tensor_shape_test.cu

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/tensor_shape.h"
+#include "dali/kernels/tensor_shape_print.h"
+#include "dali/test/device_test.h"
+
+namespace dali {
+template <int N>
+__device__ DeviceString dev_to_string(const dali::kernels::TensorShape<N> &s) {
```

Comment:
move to utils?

Reply:
Nah, the whole DeviceString thing is inefficient and really meant only for testing... Otherwise I'd have to add capacity and vector-like reallocation, which would complicate things.